Content.Server/Disease/DiseaseDiagnosisSystem.cs

```diff
@@ -0,0 +1,418 @@
+                return;
+
+            if (!TryComp<DiseaseSwabComponent>(args.Used, out var swab) || swab.Disease == null || !swab.Disease.Infectious)
+            {
+                _popupSystem.PopupEntity(Loc.GetString("diagnoser-cant-use-swab", ("machine", uid), ("swab", args.Used)), uid, Filter.Entities(args.User));
+                return;
+            }
+            _popupSystem.PopupEntity(Loc.GetString("diagnoser-insert-swab", ("machine", uid), ("swab", args.Used)), uid, Filter.Entities(args.User));
+            var machine = Comp<DiseaseMachineComponent>(uid);
+            machine.Disease = swab.Disease;
+            EntityManager.DeleteEntity(args.Used);
+
+            AddQueue.Enqueue(uid);
+            UpdateAppearance(uid, true, true);
+        }
+
+        /// <summary>
+        /// This handles swab examination text
+        /// so you can tell if they are used or not.
+        /// </summary>
+        private void OnExamined(EntityUid uid, DiseaseSwabComponent swab, ExaminedEvent args)
+        {
+            if (args.IsInDetailsRange)
+            {
+                if (swab.Used)
+                    args.PushMarkup(Loc.GetString("swab-used"));
+                else
+                    args.PushMarkup(Loc.GetString("swab-unused"));
+            }
+        }
+
+        ///
+        /// Helper functions
+        ///
+
+        /// <summary>
+        /// This assembles a disease report
+        /// With its basic details and
+        /// specific cures (i.e. not spaceacillin).
+        /// The cure resist field tells you how
+        /// effective spaceacillin etc will be.
+        /// </summary>
+        private string AssembleDiseaseReport(DiseasePrototype disease)
+        {
+            string report = string.Empty;
+            report += Loc.GetString("diagnoser-disease-report-name", ("disease", disease.Name));
+            report += System.Environment.NewLine;
+
+            if (disease.Infectious)
+            {
+                report += Loc.GetString("diagnoser-disease-report-infectious");
+                report += System.Environment.NewLine;
+            } else
+            {
+                report += Loc.GetString("diagnoser-disease-report-not-infectious");
+                report += System.Environment.NewLine;
+            }
+
+            if (disease.CureResist <= 0)
+            {
+                report += Loc.GetString("diagnoser-disease-report-cureresist-none");
+            } else if (disease.CureResist <= 0.05)
+            {
+                report += Loc.GetString("diagnoser-disease-report-cureresist-low");
+            } else if (disease.CureResist <= 0.14)
+            {
+                report += Loc.GetString("diagnoser-disease-report-cureresist-medium");
+            } else
+            {
+                report += Loc.GetString("diagnoser-disease-report-cureresist-high");
+            }
```

Comment:
```suggestion
            if (disease.CureResist <= 0)
            {
                report += Loc.GetString("diagnoser-disease-report-cureresist-none");
            } 
            else if (disease.CureResist <= 0.05)
            {
                report += Loc.GetString("diagnoser-disease-report-cureresist-low");
            } 
            else if (disease.CureResist <= 0.14)
            {
                report += Loc.GetString("diagnoser-disease-report-cureresist-medium");
            } 
            else
            {
                report += Loc.GetString("diagnoser-disease-report-cureresist-high");
            }
```

convention

or this if you're feeling fancy

```suggestion
            report += disease.CureResist switch
            {
                < 0 => Loc.GetString("diagnoser-disease-report-cureresist-none"),
                <= 0.05 => Loc.GetString("diagnoser-disease-report-cureresist-low"),
                <= 0.14 => Loc.GetString("diagnoser-disease-report-cureresist-medium"),
                _ => Loc.GetString("diagnoser-disease-report-cureresist-high")
            };
```